---
layout: page
title: LDAK-KVIK
permalink: /
---

# Welcome to LDAK-KVIK

**LDAK-KVIK** is a powerful and computationally efficient method for mixed-model association analysis in genome-wide association studies (GWAS). It is part of the [LDAK](http://www.dougspeed.com) software, which is written in C. LDAK-KVIK has been published in [Nature Genetics](https://www.nature.com/articles/s41588-025-02286-z).

### üöÄ Why use LDAK-KVIK?
 - **Scalability**: Analyzes hundreds of thousands of individuals within hours
 - **Statistical power**: Improved detection of associated loci using an elastic net model
 - **Handles structure**: Robust to relatedness and population stratification
 - **Saddlepoint approximation**: Controls type 1 error when analyzing rare diseases or skewed quantitative traits
 - **Gene-based testing**: Leverage summary statistics in a gene-based association test

To download LDAK, please visit our [downloads](/docs/downloads) page. Example code of LDAK-KVIK can be found on the [example code](/docs/example) page. An overview of the steps involved in the LDAK-KVIK algorithm is included on the [LDAK-KVIK steps](/docs/assoc/singlesnp) page.

Stay updated with the latest developments regarding LDAK software by signing up for the [LDAK mailing list](https://dougspeed.com/downloads/).

### üîß Example code

**LDAK-KVIK** is run in three steps:

1. Compute polygenic risk scores (PRS) using an elastic net model
2. Run single-SNP association analysis using PRS as an offset
3. Conduct gene-based analysis using summary statistics

Example command lines are:

```
./ldak6.linux --kvik-step1 kvik --bfile data --pheno phenofile --covar covfile --max-threads 4
./ldak6.linux --kvik-step2 kvik --bfile data --pheno phenofile --covar covfile --max-threads 4
./ldak6.linux --kvik-step3 kvik --bfile data --genefile genefile --max-threads 4
```

The main output files are:
 -  `kvik.step2.assoc` (results from single-SNP association analysis)
 -  `kvik.step3.remls.all` (results from gene-based association analysis)

### üíª Getting started

LDAK is available for Linux and Mac. On the [Downloads](/docs/downloads) page, we list several options for loading or installing the LDAK software. Explore [example code](/docs/example) and a detailed overview of the [KVIK algorithm](/docs/technical) to get started.

<!-- 
It is also possible to install the Linux-version of LDAK via [conda](http://anaconda.org/genomedk/ldak5) using the following command:
```
conda install genomedk::ldak5
```
-->
### ‚ùì Questions or issues

If you have questions or run into issues, please [raise an issue on our GitHub](https://github.com/dougspeed/LDAK/issues).